### PR TITLE
HTML + CSS + overhaul performance 

### PR DESCRIPTION
**Pull request by Thomasorus**
TLDR:
- Design is the same except when accessibility mattered most
- HTML was heavily tweaked in some spaces for accessibility and semantic reasons, but the design staus the same
- CSS was remade from scratch but old files are in the "css_old" folder
- Loading performances were increased by using woff2 instead of ttf

Notable changes:
- A reset was added to limit the number of basic CSS fixes or addons to add. It's placed inside the <head>
- For accessibility reasons, an input MUST have a label. A label was added to the search input. Not very esthetic I know, but well...
- For accessibility reasons, each page MUST start with at least an h1. It was added in some pages. In other places, h2 were converted to simple text.
- The CSS is loosely following the CUBE CSS methodology, could maybe use some cleaning.
- Entries are now lists for accessibility reasons. Screen readers announce the number of elements inside a list when they enter them, and shortcuts help jumping from one to the next

The commits can globally tell my though process when I coded this. If you need tweaks or corrections please ask.

### CODE REVIEW

**Comment by cblgh** [comment]
@Thomasorus oh wow this is MASSIVE!! thank you :>

can you post some screenshots of what it looks like atm? i'll look through the changelog later today and see if i have any questions 🖤

**Comment by cblgh** [comment]
(btw i will be squashing this when i merge, hope that's okay with you!!)

**Comment by Thomasorus** [comment]
Yeah please squash them, I've made so many so you can follow what changes were done, if you want it.

Here are the screenshots:

Capture d’écran 2021-05-14 à 12 19 01
Capture d’écran 2021-05-14 à 12 19 07
Capture d’écran 2021-05-14 à 12 19 13

**Comment by Thomasorus** [comment]
I just realized in the entries list is probably not very accessible. Since the text of the link is the url, it doesn't say much for screen readers users. And putting the description after the link would make screen readers users go back and forth. It's usable, but not ideal, your call.

**Comment by cblgh** [comment]
is it possible to have the form labels in the dom, for the benefit of screen readers, and hide them visually?

> And putting the description after the link would make screen readers users go back and forth. It's usable, but not ideal, your call.

i think that's alright for now—is there any attribute that i could add to the links that might be read by screen readers `<a title="">`? the link titles are also somewhat configurable when generating the results, so if someone really really wanted to have descriptive titles rather than the links, it's possible :)

**Comment by Thomasorus** [comment]
> is it possible to have the form labels in the dom, for the benefit of screen readers, and hide them visually?

Yes it's possible but considered bad practice to hide things. Some sighted people use screen readers too and will hear something they can't see. That's the same reason why it's bad to flip the order of things visually with CSS.

A good middle ground would be to hide it on the home, since the message right before it suggests it's a search engine, but keeping it on the results page, since there's nothing that says what the input is about even for sighted users.

> is there any attribute that i could add to the links that might be read by screen readers

Maybe we could add an `id` on each paragraph, and then use the `aria-describedby` attribute on the link. I never tested such a setup before so I don't know if screen readers will take it well or not, but in theory it could work.

Do you want me to do a second pull request for these changes?

**Comment by cblgh** [comment]
> A good middle ground would be to hide it on the home, since the message right before it suggests it's a search engine, but keeping it on the results page, since there's nothing that says what the input is about even for sighted users.

yes! it was mainly the index page that was bothering me :)

> Maybe we could add an id on each paragraph, and then use the **aria-describedby** attribute on the link. I never tested such a setup before so I don't know if screen readers will take it well or not, but in theory it could work.

would it look like something like this?
```html
<a href="https://mmm.page" id="link-0">mmm.page</a>
<p aria-describedby="link-0">Dead simple, drag & drop personal websites</p>
```

> Do you want me to do a second pull request for these changes?

would you mind adding a commit to hide the form label on the index page? and i can take care of the aria stuff (if my snippet above is correct)

really, really appreciate yr work! kinda blown away how you did such a revamp and it looks... the same?! merci beaucoup doctor, vraiment :3

**Comment by Thomasorus** [comment]
It's the opposite :+1: 
```html
<a href="https://mmm.page" aria-describedby="link-0">mmm.page</a>
<p  id="link-0">Dead simple, drag & drop personal websites</p>
```
The `a` is described by the `p`. Of course you need a different `id` for each entry.

For hiding the label I already knew we were going to talk about it so I prepared something special! :smiling_imp:  You already have a class named `visually-hidden` in the CSS. Just add it to the label in the index.html

```html
<label for="search" class="visually-hidden">Search {{ .SiteName }}</label>
```
This way screen readers will announce it, but sighted users won't see it. :fist_raised:

**Comment by cblgh** [comment]
(i'll merge this after i have read through all of the commits. learning so much rn 🤩)

**Comment by Thomasorus** [comment]
If you need explanations about them please ask. I was almost thinking we could record a video following the commits. 😂

**Comment by cblgh** [comment]
thank you so much @Thomasorus!! i have a list of things i'll be looking up; i'll ping you on the masto if i don't get the explanation of something :3

**Comment by rosano** [comment]
that's a massive collab, congrats to both of you! 🎉
